BazelExtensions/xcode_configuration_provider.bzl

```diff
@@ -313,6 +322,9 @@ def _source_output_file_map(target, ctx):
         if len(srcs) != len(objs):
             fail("[ERROR] Unexpected number of object files")
         for src in srcs:
+            # Skip sources that are not relevant for the Xcode experience
+            if src.path.count("bazel-out") > 0 or src.path.startswith("external/"):
```

Comment:
XCHammer generally works with generated files in bazel-out and external - it could cause issues for those paths right?

The other large issue I noticed was this line:
```
    implementation=_xcode_build_sources_aspect_impl, attr_aspects=["*"],
```

Concretely - I wonder if we can atleast narrow this down - it's probably a harder change
```
attr_aspects
```

Reply:
Yeah, the number of files generated influence how much memory `xcbuildkit` needs to process and load mapping info from these files. That's why I was exploring the idea of reducing this somehow.

Let me try to narrow the attributes being visited as you suggested and see how it goes. Will hold this PR for now, thx for the tip!